website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,237 @@
+                    ('active', '=', True)
+                    ])
+            return request.render(
+                'website_chd_product_configurator.configurator', {
+                    'curr_product_id': curr_product_id,
+                    'curr_types': curr_types,
+                    'avail_accessories': avail_accessories,
+                    'curr_chd_price_component_ats':
+                    curr_chd_price_component_ats,
+                    'errormsg': errormsg,
+                    })
+
+        return request.render(
+            'website_chd_product_configurator.sale_options',
+            {
+                'curr_product_id': curr_product_id,
+                'curr_chd': new_chd,
+                'all_accessories': all_accessories,
+                'results': results,
+                'configuration_form': str(form_data),
+                'message': message,
+                'summary': form_data['summary'],
+                })
+
+    def get_current_partner(self):
+        partner_model = request.env['res.partner']
+        current_partner = partner_model.search(
+            [('user_account_id', '=', request.uid)])
+        if len(current_partner) == 0:
+            return False
+        else:
+            return current_partner[0]
+
+    @http.route('/chd_init/buy<id>/', website=True)
+    def chosen_option(self, **form_data):
+        partner = self.get_current_partner()
+        if partner is False:
+            return request.render(
+                'website_chd_product_configurator.no_partner', {})
+        result_model = request.env['chd.product_configurator.result']
+        result = result_model.search([('id', '=', form_data['id'])])
+        configurator = request.env['chd.product_configurator'].search(
+            [('id', '=', result.configurator_id.id)])
+        request.context['active_id'] = result.id
+        fields = [
+            'order_id',
+            'return_to_order',
+            'display_order_id',
+            'result_id']
+        if form_data['action'] == 'buy':
+            doorder_model = request.env['chd.product_configurator.do_order']
+            # again, access 7.0 with ._model property
+            doorder_res = doorder_model._model.default_get(
+                request.cr, request.uid,
+                fields_list=fields, context=request.context)
+            #will make all "erased" results point to wishlist 0.
+            result.write({
+                'wishlist': 0,
+                })
+            order = request.env['sale.order'].search(
+                [('id', '=', doorder_res['order_id'])])
+            return request.render(
+                'website_chd_product_configurator.buy_option', {
+                    'summary': result.summary,
+                    'result': result,
+                    'order': order,
+                    })
+
+    # method for fetching finishing options via json
+    @http.route('/chd_init/getch/', type='json')
+    def tr(self, type_id):
```

Comment:
You don't need a controller for this. You can declare this function on your model and call that from the javascript side. But in this case, you can do all the business logic in javascript
```
(new openerp.web.Model('product.finishing'))
.query(['id', 'name'])
.filter([['type_option_ids', 'in', /*wherever your type_id comes from*/]])
.all()
.then(function(results)
{
    //do something with the results
})
```